lib/patches/mailer_patch.rb

```diff
@@ -0,0 +1,79 @@
+require_dependency 'mailer'
+
+module Patches
+  module MailerPatch
+    def self.included(base)
+      base.send(:include, InstanceMethods)
+      base.class_eval do
+        alias_method :super_issue_edit, :issue_edit
+        alias_method :issue_edit, :patched_issue_edit
+
+        alias_method :super_message_posted, :message_posted
+        alias_method :message_posted, :patched_message_posted
+      end
+    end
+
+    module InstanceMethods
+      def patched_issue_edit(user, journal)
```

Comment:
I'm thinking that a better solution than rewrite the entire function, is to just rewrite the view, and do you modifications there.
If needed, you can also overwrite the function, but just set some instance variables (`@..`) to be accessed in view (if needed). Since the view is and `erb` you can also write ruby there.

Unfortunatally I don't have an example for how to do it. Is possible that when writing a view with the same name in your plugin it just overwrites the default one (from redmine)? You will have to try and see.

Reply:
Thanks for your input! I’ve made the other changes, including the refactoring and internationalization, and committed them.
As for your suggestion to rewrite the view instead of modifying the function, I see your point. Following this approach, I’ll likely need to remove my mailer_patch, which is fine. I’m working on trying this out now and will keep you updated on how it goes.
Let me know if you have any additional thoughts as I proceed with this change.

Reply:
I've made the changes you suggested. I removed the logic from the mailer patch and moved it directly into the views. The view now handles whether to show the teaser or the full note based on the setting `notifications_mail_option`. Everything seems to be working well without the need for extra instance variables or patches.